From b95544e93c27660fe8875787ee9358b64639ae0b Mon Sep 17 00:00:00 2001
From: Hugues Fruchet <hugues.fruchet@foss.st.com>
Date: Mon, 19 Feb 2024 16:25:56 +0100
Subject: [PATCH] media: verisilicon: h1: add RGB support for STM32MP25 VENC

Add support of encoding from RGB.
RGB565 16 bits and RGB888 32bits variants are supported.

Signed-off-by: Hugues Fruchet <hugues.fruchet@foss.st.com>
Change-Id: I93cbf370e448dea9959446b42040a99e63e62660
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/336821
Reviewed-by: Philippe CORNU <philippe.cornu@foss.st.com>
Reviewed-by: Hugues FRUCHET <hugues.fruchet@st.com>
Domain-Review: Philippe CORNU <philippe.cornu@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Tested-by: Hugues FRUCHET <hugues.fruchet@st.com>
---
 .../platform/verisilicon/stm32mp25_vpu_hw.c   | 35 +++++++++++++++++++
 1 file changed, 35 insertions(+)

--- a/drivers/media/platform/verisilicon/stm32mp25_vpu_hw.c
+++ b/drivers/media/platform/verisilicon/stm32mp25_vpu_hw.c
@@ -79,6 +79,41 @@ static const struct hantro_fmt stm32mp25
 		.enc_fmt = ROCKCHIP_VPU_ENC_FMT_UYVY422,
 	},
 	{
+		.fourcc = V4L2_PIX_FMT_RGB565,
+		.codec_mode = HANTRO_MODE_NONE,
+		.enc_fmt = ROCKCHIP_VPU_ENC_FMT_RGB565,
+	},
+	{
+		.fourcc = V4L2_PIX_FMT_XBGR32,
+		.codec_mode = HANTRO_MODE_NONE,
+		.enc_fmt = ROCKCHIP_VPU_ENC_FMT_RGB888,
+	},
+	{
+		.fourcc = V4L2_PIX_FMT_BGR32,
+		.codec_mode = HANTRO_MODE_NONE,
+		.enc_fmt = ROCKCHIP_VPU_ENC_FMT_RGB888,
+	},
+	{
+		.fourcc = V4L2_PIX_FMT_RGBX32,
+		.codec_mode = HANTRO_MODE_NONE,
+		.enc_fmt = ROCKCHIP_VPU_ENC_FMT_RGB888,
+	},
+	{
+		.fourcc = V4L2_PIX_FMT_BGRX32,
+		.codec_mode = HANTRO_MODE_NONE,
+		.enc_fmt = ROCKCHIP_VPU_ENC_FMT_RGB888,
+	},
+	{
+		.fourcc = V4L2_PIX_FMT_XRGB32,
+		.codec_mode = HANTRO_MODE_NONE,
+		.enc_fmt = ROCKCHIP_VPU_ENC_FMT_RGB888,
+	},
+	{
+		.fourcc = V4L2_PIX_FMT_RGB32,
+		.codec_mode = HANTRO_MODE_NONE,
+		.enc_fmt = ROCKCHIP_VPU_ENC_FMT_RGB888,
+	},
+	{
 		.fourcc = V4L2_PIX_FMT_JPEG,
 		.codec_mode = HANTRO_MODE_JPEG_ENC,
 		.max_depth = 2,
